### PR TITLE
https://github.com/tgstation/tgstation/pull/65189

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
- https://github.com/tgstation/tgstation/pull/65189
- https://github.com/tgstation/tgstation/pull/67976
- https://github.com/tgstation/tgstation/pull/71060

## Changelog

:cl:
add: Multi-Z now supports things such as med huds and antag huds
add: Alot of animations now play across Z levels
/:cl:

### CODE REVIEW

File: code/game/alternate_appearance.dm [hunk 1]
```diff
@@ -72,43 +88,62 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
...
+
 /datum/atom_hud/alternate_appearance/basic/Destroy()
 	. = ..()
 	QDEL_NULL(image)
 	target = null
 	if(ghost_appearance)
 		QDEL_NULL(ghost_appearance)
 
-/datum/atom_hud/alternate_appearance/basic/add_to_hud(atom/A)
+/datum/atom_hud/alternate_appearance/basic/proc/mimic(atom/movable/openspace/mimic)
```
Comment (Lohikar) [inline code/game/alternate_appearance.dm:115]: Nit: `/atom/movable/openspace/mimic/M` or something, the /openspace type is abstract and not relevant here.

File: code/modules/antagonists/_common/antag_hud.dm [hunk 2]
```diff
@@ -49,6 +49,9 @@ GLOBAL_LIST_EMPTY_TYPED(has_antagonist_huds, /datum/atom_hud/alternate_appearanc
 
 	return ..()
 
+/datum/atom_hud/alternate_appearance/basic/antagonist_hud/mimic(atom/movable/openspace/mimic)
```
Comment (Lohikar) [inline code/modules/antagonists/_common/antag_hud.dm:52]: Here as well.

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -105,24 +105,97 @@
...
+ */
+/atom/proc/set_hud_image_active(hud_category, update_huds = TRUE, datum/atom_hud/exclusive_hud)
+	if(!istext(hud_category) || !hud_list?[hud_category] || active_hud_list?[hud_category])
+		return FALSE
+
+	LAZYSET(active_hud_list, hud_category, hud_list[hud_category])
+
+	if(ismovable(src))
+		var/atom/movable/AM = src
+		for(var/atom/movable/mimic as anything in AM.get_associated_mimics())
```
Comment (Lohikar) [inline code/modules/mob/mob.dm:124]: You could use `FOR_MIMIC_OF` here if you want to microoptimize a bit.

File: code/game/alternate_appearance.dm (hunk 1 above)
Comment (Kapu1178) [reply code/game/alternate_appearance.dm:115]: Unintentional, was meant to be typed to mimic.

**Pull request merged**